crates/cranelift/src/func_environ.rs

```diff
@@ -766,9 +766,9 @@ impl<'module_environment> FuncEnvironment<'module_environment> {
         // if null, we take a slow-path that invokes a
         // libcall.
         let table_entry_addr = builder.ins().table_addr(pointer_type, table, index, 0);
-        let value = builder
-            .ins()
-            .load(pointer_type, ir::MemFlags::trusted(), table_entry_addr, 0);
+        let mut flags = ir::MemFlags::trusted();
+        flags.set_table();
+        let value = builder.ins().load(pointer_type, flags, table_entry_addr, 0);
```

Comment:
These are becoming very verbose. Can we make `set_table` et al take and return `Self` so that we can chain the methods like

```rust
ir::MemFlags::trusted().set_table()
```

Reply:
Sure, good idea; I added `with_<flag>` methods to `MemFlags`.